python/sglang/srt/mem_cache/memory_pool.py

```diff
@@ -466,6 +468,333 @@ def move_kv_cache(self, tgt_loc: torch.Tensor, src_loc: torch.Tensor):
+                        loc = torch.arange(self.k.shape[0], device=self.outer_self.device)
+                        self.outer_self.gpu_cache.set_kv_buffer(
+                            layer=None,
+                            loc=loc,
+                            cache_k=k_gpu,
+                            cache_v=v_gpu,
+                            layer_id_override=self.layer_id
+                        )
+
+                    self.event.set()
+                except Exception as e:
+                    print(f"Layer {self.layer_id} upload failed: {str(e)}")
+                    self.event.set()
+
+        # Create and submit task with reference to self
+        task = UploadTask(self, layer_id, k, v)
+
+        if high_priority:
+            self.prefetch_executor.submit(task)
+        else:
+            self.prefetch_executor.submit(task)
+
+        return task.event
+
+    def _async_prefetch(self, layer_id: int):
+        """Thread-safe prefetch mechanism"""
+        if layer_id < 0 or layer_id >= self.layer_num:
+            return
+
+        with self.prefetch_lock:
+            if layer_id in self.prefetch_queue:
+                return
+            self.prefetch_queue.add(layer_id)
+            self.stats.increment('prefetches')
+
+        def prefetch_task():
+            try:
+                # Load from disk
+                k_disk, v_disk = self.disk_cache.get_kv_buffer(layer_id)
+
+                # Upload to GPU
+                self._async_upload_to_gpu(layer_id, k_disk, v_disk)
+            finally:
+                with self.prefetch_lock:
+                    self.prefetch_queue.discard(layer_id)
+
+        self.prefetch_executor.submit(prefetch_task)
+
+    def backup_to_host_all_layer(self) -> int:
+        """
+        Backup all layers from GPU to disk
+        Returns number of successfully backed up layers
+        """
+        print(f"backup_to_host_all_layer")
+        success_count = 0
+        futures = []
+
+        with ThreadPoolExecutor(max_workers=4) as executor:
+            for layer_id in range(self.layer_num):
+                futures.append(executor.submit(self._backup_single_layer, layer_id))
+
+            for future in futures:
+                if future.exception() is None and future.result():
+                    success_count += 1
+                elif future.exception():
+                    self.stats.increment('errors')
+
+        return success_count
+
+    def load_from_host_per_layer(self, layer_id: int) -> bool:
+        """
+        Fixed version that properly handles pinned memory
+        """
+        print(f"load from host per layer {layer_id}")
+        try:
+            if layer_id < 0 or layer_id >= self.layer_num:
+                raise ValueError(f"Invalid layer_id: {layer_id}")
+
+            k_disk, v_disk = self.disk_cache.get_kv_buffer(layer_id)
+
+            if k_disk is None or v_disk is None:
+                raise RuntimeError(f"Disk data corrupted for layer {layer_id}")
```

Comment:
Is this condition always `False`? The function ` _initialize_layer_file` in `DiskKVCache` will be called to create the pre-allocated file at the beginning, and k_disk and v_disk always exist even though KV haven't been copied to disk yet. Should we check if the files on the disk really includes the KV cache we need?